### PR TITLE
Modifies README and adds Getting Started doc.

### PR DESCRIPTION
**Pull request by jvrahav**
This change modifies readme, getting started doc and an uber deployment spec file which can be used to deploy the operator

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR brings in minor documentation changes

### CODE REVIEW

**Comment by asifdxtreme** [comment]
@jvrahav 
We are getting error in vdo-controller pod, here are the error details :
```
vmware-system-vdo    vdo-controller-manager-5f865cb8c7-95xfk      1/2     ImagePullBackOff   0          8s
```

```
vents:
1892383
  Type     Reason     Age              From               Message
1892384
  ----     ------     ----             ----               -------
1892385
  Normal   Scheduled  8s               default-scheduler  Successfully assigned vmware-system-vdo/vdo-controller-manager-5f865cb8c7-95xfk to kind-control-plane
1892386
  Normal   Pulling    7s               kubelet            Pulling image "gcr.io/kubebuilder/kube-rbac-proxy:v0.5.0"
1892387
  Normal   Pulled     5s               kubelet            Successfully pulled image "gcr.io/kubebuilder/kube-rbac-proxy:v0.5.0" in 1.890665532s
1892388
  Normal   Created    4s               kubelet            Created container kube-rbac-proxy
1892389
  Normal   Started    4s               kubelet            Started container kube-rbac-proxy
1892390
  Normal   Pulling    4s               kubelet            Pulling image "vmware.com/vdo:40a4440"
1892391
  Warning  Failed     4s               kubelet            Failed to pull image "vmware.com/vdo:40a4440": rpc error: code = Unknown desc = failed to pull and unpack image "vmware.com/vdo:40a4440": failed to resolve reference "vmware.com/vdo:40a4440": failed to do request: Head "https://vmware.com/v2/vdo/manifests/40a4440": x509: certificate signed by unknown authority
1892392
  Warning  Failed     4s               kubelet            Error: ErrImagePull
1892393
  Normal   BackOff    2s (x2 over 3s)  kubelet            Back-off pulling image "vmware.com/vdo:40a4440"
1892394
  Warning  Failed     2s (x2 over 3s)  kubelet            Error: ImagePullBackOff
```

File: artifacts/vdo-spec.yaml [hunk 1]
```diff
@@ -0,0 +1,673 @@
...
+            configMapKeyRef:
+              key: versionConfigURL
+              name: compat-matrix-config
+              optional: true
+        - name: AUTO_UPGRADE
+          valueFrom:
+            configMapKeyRef:
+              key: auto-upgrade
+              name: compat-matrix-config
+        image: vmware.com/vdo:40a4440
```
Comment (asifdxtreme) [inline artifacts/vdo-spec.yaml:625]: This ID is getting used to deploy vdo-controller, however the make deploy makes the docker image with a different tag at runtime, we need to update this spec.yaml with the new tag

**Review by asifdxtreme — APPROVED** [review]
LGTM

**Review by ridaz — APPROVED** [review]
LGTM, Thanks.

**Pull request merged**